File: dotnet/niuz.application.test/scenarios/PayByPublication.cs
using FluentAssertions;
using niuz.application.dtos;
using niuz.application.fixtures;
using niuz.application.website;
using Xunit;

namespace niuz.application.scenarios
{
    public class PayByPublication
    {
        private readonly TestFacade app;

        public PayByPublication()
        {
            app = new TestFacade();
        }

        [Fact]
        public void SubmitArticle()
        {
            app.Hire("author-1", "Freddy Kruger", "123-4567-89", "pay-by-publication", 100);
            app.Submit("article-1", "author-1", "headline");

            app.Get("homepage").Should().BeEmpty();
            app.GetByBankAccount("123-4567-89").Should().BeEmpty();
        }

        [Fact]
        public void PublishArticle()
        {
            app.Hire("author-1", "Freddy Kruger", "123-4567-89", "pay-by-publication", 100);
            app.Submit("article-1", "author-1", "headline");
            app.Publish("article-1");

            app.Get("homepage").Should().ContainInOrder(
                new TeaserDto("headline", "Freddy Kruger")
            );
            app.GetByBankAccount("123-4567-89").Should().ContainInOrder(
                new PaymentDto(100, "123-4567-89", "Freddy Kruger", "headline")
            );
        }
    }
}

File: java/src/test/java/com/dddeurope/niuz/scenarios/PayByPublication.java
package com.dddeurope.niuz.scenarios;

import com.dddeurope.niuz.finance.PaymentDto;
import com.dddeurope.niuz.hr.HireAuthor;
import com.dddeurope.niuz.newsroom.PublishArticle;
import com.dddeurope.niuz.newsroom.SubmitArticle;
import com.dddeurope.niuz.website.TeaserDto;
import com.dddeurope.niuz.fixtures.TestFacade;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class PayByPublication {
    private TestFacade app;

    @BeforeEach
    void setUp() {
        app = new TestFacade();
    }

    @Test
    void submitArticle() {
        app.dispatch(new HireAuthor("author-1", "Freddy Kruger", "pay-by-publication", 100, "123-4567-89"));
        app.dispatch(new SubmitArticle("article-1", "author-1", "headline"));

        assertThat(app.get("homepage")).isEmpty();
        assertThat(app.getByBankAccount("123-4567-89")).isEmpty();
    }

    @Test
    void publishArticle() {
        app.dispatch(new HireAuthor("author-1", "Freddy Kruger", "pay-by-publication", 100, "123-4567-89"));
        app.dispatch(new SubmitArticle("article-1", "author-1", "headline"));
        app.dispatch(new PublishArticle("article-1"));

        assertThat(app.get("homepage")).containsExactly(
                new TeaserDto("headline", "Freddy Kruger")
        );
        assertThat(app.getByBankAccount("123-4567-89")).containsExactly(
                new PaymentDto(100, "123-4567-89", "Freddy Kruger", "headline")
        );
    }
}
